src/utils/transaction_builder_config.ts

```diff
@@ -0,0 +1,108 @@
+import { C, SlotConfig } from "../mod.ts";
+import { ProtocolParameters } from "../types/mod.ts";
+import { createCostModels } from "./cost_model.ts";
+import { Freeable, Freeables } from "./freeable.ts";
+
+export function getTransactionBuilderConfig(
+  protocolParameters: ProtocolParameters,
+  slotConfig: SlotConfig,
+  blockfrostConfig: {
+    url?: string;
+    projectId?: string;
+  },
+) {
+  const bucket: Freeable[] = [];
+  let builderA = C.TransactionBuilderConfigBuilder.new();
+
+  const coinsPerUtxoByte = C.BigNum.from_str(
+    protocolParameters.coinsPerUtxoByte.toString(),
+  );
+  bucket.push(coinsPerUtxoByte);
+  let builderB = builderA.coins_per_utxo_byte(coinsPerUtxoByte);
+  builderA.free();
+
+  const minFeeA = C.BigNum.from_str(protocolParameters.minFeeA.toString());
+  bucket.push(minFeeA);
+  const minFeeB = C.BigNum.from_str(protocolParameters.minFeeB.toString());
+  bucket.push(minFeeB);
+  const linearFee = C.LinearFee.new(minFeeA, minFeeB);
+  bucket.push(linearFee);
+  builderA = builderB.fee_algo(linearFee);
+  builderB.free();
```

Comment:
you can push this. I mean what you do is fine but if you add to the bucket there is no issue

Reply:
Each time we call one of the builder functions, it clones the previous builder and returns a new one. If I were to push the value to the bucket and free `finally`, it would leave lots of dangling references.